app/Http/Controllers/Api/AssetsController.php

```diff
@@ -706,8 +706,11 @@ public function checkin(Request $request, $asset_id)
         $asset->assigned_to = null;
         $asset->assignedTo()->disassociate($asset);
         $asset->accepted = null;
-        $asset->name = Input::get('name');
-        $asset->location_id =  $asset->rtd_location_id;
+        $asset->location_id = $asset->rtd_location_id;
+
+        if (Input::has('name')) {
```

Comment:
We should probably use the more conventional version here without the Input:: helper:

```php
if ($request->filled('name')) {
    $asset->name = Input::get('name');
}
```

Input isn't used in Snipe-IT as much anymore and is more related to form submissions. It was an oversight that it's still used here in the API.